pkg/pipeline/PipelineStageService.go

```diff
@@ -2025,3 +2093,96 @@ func (impl *PipelineStageServiceImpl) BuildPluginVariableAndConditionDataForWfRe
 }
 
 //BuildPrePostAndRefPluginStepsDataForWfRequest and related methods ends
+
+func (impl *PipelineStageServiceImpl) fetchScopedVariablesAndResolveTemplate(unresolvedResponse *bean.PrePostAndRefPluginStepsResponse, pipelineStageIds []int, scope models.Scope) (*bean.PrePostAndRefPluginStepsResponse, error) {
+
+	entities := make([]repository3.Entity, 0)
+	for _, stageId := range pipelineStageIds {
+		entities = append(entities, repository3.Entity{
+			EntityType: repository3.EntityTypePipelineStage,
+			EntityId:   stageId,
+		})
+	}
+	mappingsForEntities, err := impl.variableEntityMappingService.GetAllMappingsForEntities(entities)
```

Comment:
we should add a check in `variableEntityMappingService` if entities are empty, we should not make db call